Libraries/LibGUI/GTextEditor.cpp

```diff
@@ -932,23 +952,24 @@ void GTextEditor::update_cursor()
 
 void GTextEditor::update_undo_timer()
 {
-    if (m_undo_stack.size() <= 0)
+    if (m_undo_stack.is_empty())
         return;
 
-    if (m_undo_timer == 0)
-        m_prev_undo_stack_size = m_undo_stack[m_undo_index].size();
+    auto& undo_vector = m_undo_stack[0].m_undo_vector;
 
-    if (m_undo_timer >= 2 && m_undo_stack[m_undo_index].size() > 0) {
+    if (undo_vector.size() == m_last_updated_undo_vector_size && !undo_vector.is_empty()) {
+        auto undo_commands_container = make<UndoCommandsContainer>();
+        m_undo_stack.insert(0, move(undo_commands_container));
 
-        if (m_undo_stack[m_undo_index].size() == m_prev_undo_stack_size) {
-            dbg() << "Increased Undo Index";
-            m_undo_stack.append(make<NonnullOwnPtrVector<UndoCommand>>());
-            m_undo_index++;
-        }
-        m_undo_timer = -1;
+        // Note: Remove dbg() if were 100% sure there are no bugs left.
```

Comment:
Typo: we're.
Also, I love the implication that we'll ever be 100% sure of that :)